proposed/0000-name-service.md

```diff
@@ -0,0 +1,67 @@
+- Feature Name: Decentralised Naming System (dns)
+- Type new feature
+- Related components maidsafe_client, maidsafe_vault
+- Start Date: 27-06-2015
+- RFC PR:
+- Issue number: 
+
+# Summary
+
+In the current Internet Domains are referenced via a naming system which comprises an indexing mechanism
+This indexing mechanism is centralised and controlled, with the ability for countries and ISPs to switch it off
+pollute data with advertising data and much worse. The SAFE network on the other had offers a 'free (as in beer)'
+mechanism to locate data linked to a name. This proposal outlines a mechanism to look up data related to any name.
+
+This data includes, long name (could be real name is users wish), web site id, blog id, micro-blog id and may be 
+extended to include additional information as the network develops. 
+
+# Motivation
+
+In networks people expect to be able to lookup services related to names. In the SAFE network this includes
+the ability to retrieve public keys to encrypt data to that id, in cases where privacy is a requirement (.e.
+in SAFE all messages are encrypted between identities). The opportunity for people to create and link web sites
+and other services is also a motivation for implementing such a system. 
+
+# Detailed design
+
+This is a simple use case for `Unified Structured Data` (see RFC XXXX). In this case we require to set three items
+
+1. The `type_tag` shall be type `4`
+
+2. The `Identity` shall be the `Sha512` of the publicly chosen name (user chooses this name)
+
+3. The `data` field:
+
+```rust
+struct dns {
+long_name: String
+encrytion_key: crypto::encrypt::public_key
+//      service, location
+HashMap<String, Identity>
+}
+
+Initially this `HashMap` will likely contain www, blog and micro-blog. Application developers may add services to enhance 
+users experiences with new applications and functions via this mechanism. 
+
```

Comment:
So just as example: Say I want to create 2 web-sites.
So HashMap entries will be:
`www.website0.com` => `Root Directory ID 0 that contains HOME.html etc files`
`www.website1.com` => `Root Directory ID 1 that contains HOME.html etc files`

<1> What is `long_name` in this scenario ?
<2> What is the purpose of `encryption_key` field ?

The signature of this Structured Data can/will use some completely separate Key-Pair.
<3> Do we need to restrict the number of Key-Pair the use can generate or the number of Structured Data with type_tag = 4 that user can create ?